libs/langchain/tests/unit_tests/document_loaders/parsers/test_pdf_parsers.py

```diff
@@ -53,6 +54,51 @@ def _assert_with_parser(parser: BaseBlobParser, splits_by_page: bool = True) ->
         assert int(metadata["page"]) == 0
 
 
+def _assert_with_bytes_parser(
```

Comment:
Could we pull out the dependent code into standalone function and have it all run from the same standard tester, so by default all parsers have to be verified to work with it?

Reply:
I may be misunderstanding what you are asking. I did write a standalone unit test for it. Passing the standard test should work - eg, it should be able to read the test doc correctly - but that won't test its ability to extract email-like metadata since there isn't any in the test doc. But, if for completeness, this test should be there as well, glad to add it,

Reply:
> Could we pull out the dependent code into standalone function and have it all run from the same standard tester, so by default all parsers have to be verified to work with it?

should this class be packaged in pypdf with the other parsers?